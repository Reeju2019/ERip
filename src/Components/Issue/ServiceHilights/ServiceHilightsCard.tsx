import * as React from 'react'

interface IServiceHilightsCardProps {
  item: {
    title: string
    text: string
    image: string
  }
}

const ServiceHilightsCard: React.FunctionComponent<IServiceHilightsCardProps> = ({ item }) => {
  return (
    <>
      <div className='d-flex flex-row divContainer p-2 my-4'>
        <div>
          <img src={item.image} className='itemImg' alt='itemImg' />
        </div>
        <div className='p-2'>
          <h6>{item.title}</h6>
          <p className='itemPara'>{item.text}</p>
        </div>
      </div>
    </>
  )
}

export default ServiceHilightsCard
