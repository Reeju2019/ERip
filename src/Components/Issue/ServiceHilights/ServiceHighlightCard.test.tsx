import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ServiceHilightsCard from './ServiceHilightsCard'


interface Data {
    title: string
    text: string
    image: string
}
const data: Data = {
    title: "",
    text: "",
    image: ""
}

test('service highlight card', () => {
    render(<ServiceHilightsCard item={data} />)
})
describe('service highlight card rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<ServiceHilightsCard item={data} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing service highlight card', () => {
        const div = document.createElement('div')
        render(<ServiceHilightsCard item={data} />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
