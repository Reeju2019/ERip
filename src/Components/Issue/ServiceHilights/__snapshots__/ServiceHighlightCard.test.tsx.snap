// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service highlight card rendering specification SnapShot testing 1`] = `
<div
  className="d-flex flex-row divContainer p-2 my-4"
>
  <div>
    <img
      alt="itemImg"
      className="itemImg"
      src=""
    />
  </div>
  <div
    className="p-2"
  >
    <h6 />
    <p
      className="itemPara"
    />
  </div>
</div>
`;
